### PR TITLE
Reorganize project

### PR DESCRIPTION
**Pull request by rshym**
- remove self-reference from the yaml;
- move yocto layers into the dedicated directory;
- add .gitignore with build products;
- remove 'Fetching' chapter from the README as outdated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,14 +71,12 @@ Features that are present but not tested:
...
-fetch this repository again into `yocto/` directory. So, to avoid
-possible confusion, we recommend to download only `prod-devel-rcar.yaml`:
-
-```
-# curl -O https://raw.githubusercontent.com/xen-troops/meta-xt-prod-devel-rcar/master/prod-devel-rcar.yaml
-```
+You can clone this whole repository, or download it as an archive.
+During the build few directories will be created and additional
+dependencies will be fetched into them.
+There will be created the build directory `yocto/` for yocto's
```
Comment (lorc) [inline README.md:77]: A bit strange phrasing. Maybe something like this?

"The build system will create build directory `yocto/` for yocto's
meta-layers, and `zephyr/` and/or `android/` directories,
depending on the build options."

File: README.md [hunk 2]
```diff
@@ -128,6 +127,27 @@ salvator-xs-h3-4x2g --ENABLE_ANDROID yes`.
 
 This will require even more time and space, as Android is quite big.
 
+## Build products
+
+During the build some interim and final products will appear.
```
Comment (lorc) [inline README.md:132]: Well, `build.ninja` is not a product, strictly speaking. Let's call it "artifact"

File: README.md [hunk 3]
```diff
@@ -128,6 +127,27 @@ salvator-xs-h3-4x2g --ENABLE_ANDROID yes`.
...
+
+After `moulin prod-devel-rcar.yaml`:
+```
+| buld.ninja
```
Comment (lorc) [inline README.md:136]: `build.ninja`

**Review by lorc — COMMENTED** [review]
I played a bit with this PR. It feels very nice.

There are couple of minor comments regarding documentation. Also it would be great to extend the commit message. At least write **why** you are doing this.

File: README.md (hunk 1 above)
Comment (rshym) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (rshym) [reply README.md]: Fixed. The sentence is reworded.

File: README.md (hunk 3 above)
Comment (rshym) [reply README.md]: Fixed.

**Comment by rshym** [comment]
Comments are addressed.
The commit message is provided with additional explanations.

**Review by lorc — APPROVED** [review]
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by rshym** [comment]
`Rb` is added.
Tested by the build `moulin prod-devel-rcar.yaml --MACHINE h3ulcb-4x2g --GRAPHICS binaries --ENABLE_DOMU yes`.
Ready to be merged.

**Pull request merged**